### DISCUSSION TITLE
How to properly setup/use the Copilot extra?

### DISCUSSION THREAD

**Discussion by SwiftedMind** [LazyVim/LazyVim#501] [Q&A]
Hey, I'm new to LazyVim and I absolutely love it!

I'm having some issues with getting Copilot running properly, though. I added the import to the `lazy.lua` file as stated in the documentation: https://www.lazyvim.org/plugins/extras/coding.copilot

However, there are two more sections `copilot.lua` and `copilot-cmp` and I'm not sure what I have to do with them or if that's just information.

I enabled Copilot and it does kind of work. Occasionally, I get a suggestion from "Copilot" in the suggestion list. However, it's pretty rare and I never get suggestions in a new, empty line. In VSCode, I could do something like this:

```
// Write a function that does this and that
{ empty new line and Copilot suggests the implementation}
```

This strangely doesn't work in lazyvim/neovim. When I hit enter, I get no suggestions. Is there a way to enable that? Or is that not supported?

Also, maybe related to the problem, I'm not sure: When I write a comment and then add a new line, that new line automatically becomes a comment as well, which I have to manually remove every time to get an actually empty line for Copilot. I have no idea how to turn that off. I tried putting the following in the `options.lua` file, but it didn't work:

```lua
vim.opt.formatoptions:remove({ "c", "r", "o" })
```

I'd appreciate any help with this :)

Cheers

**Comment by folke** [comment]
There's a cmp/copilot issue about the empty line completion https://github.com/zbirenbaum/copilot-cmp/issues/5

**Reply by SwiftedMind** [reply]
Ah okay, got it. Thanks!

**Comment by Louis-dM** [comment]
> However, there are two more sections copilot.lua and copilot-cmp and I'm not sure what I have to do with them or if that's just information.

Did you ever find the answer to this? I've assumed copilot.lua is a new file that should be created in lua/plugins, but I don't know about copilot-cmp. There's also a "lualine.nvim" section now and I don't know where in the config that's supposed to go either.

**Reply by folke** [reply]
You just need to add the import as explained here https://www.lazyvim.org/plugins/extras/coding.copilot
Nothing else needs to b done

**Reply by Louis-dM** [reply]
Oh okay. So what are the copilot.lua/lualine.vnvim/copilot-cmp sections for?

**Reply by folke** [reply]
That's just the documentation showing you the options that were set.

**Reply by Louis-dM** [reply]
Ohhh. Thank you :)
